packages/go_router/lib/src/go_route_information_provider.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:go_router/src/go_route_information_parser.dart';
+
+/// The route information provider created by go_router
+class GoRouteInformationProvider extends RouteInformationProvider
```

Comment:
Does this have tests? If no, why not?

Reply:
It probably should, I'll look into adding some

Reply:
Added some rudimentary tests in [1061a9c](https://github.com/flutter/packages/pull/2189/commits/1061a9cc68d2db2a1a85dd78dbf07a3f7acd8e7e)